csi/src/dev.rs

```diff
@@ -0,0 +1,175 @@
+                                                if let Some(devname) = device
+                                                    .property_value("DEVNAME")
+                                                {
+                                                    if let Some(name) =
+                                                        devname.to_str()
+                                                    {
+                                                        let value = iscsi::IscsiDetach::from_path(name.to_string(), path)?;
+                                                        if value.uuid() == uuid
+                                                        {
+                                                            return Ok(Some(
+                                                                Box::new(value),
+                                                            ));
+                                                        }
+                                                    }
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    continue;
+                }
+            }
+
+            if let Some(wwn) = device.property_value("ID_WWN") {
+                if wwn.to_str() == Some(&nvmf_key) {
+                    if let Some(model) = device.property_value("ID_MODEL") {
+                        if model.to_str() == Some("Mayastor NVMe controller") {
+                            if let Some(devname) =
+                                device.property_value("DEVNAME")
+                            {
+                                if let Some(name) = devname.to_str() {
+                                    return Ok(Some(Box::new(
+                                        nvmf::NvmfDetach::new(
+                                            name.to_string(),
+                                            format!(
+                                                "{}:nexus-{}",
+                                                NVME_NQN_PREFIX,
+                                                uuid.to_string()
+                                            ),
+                                        ),
+                                    )));
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(None)
+    }
+
+    /// Wait for a device to show up in udev
+    /// once attach() has been called.
+    pub async fn wait_for_device(
+        device: Box<dyn Attach>,
+        retries: u32,
+    ) -> Result<DeviceName, DeviceError> {
+        const SECOND: Duration = Duration::from_secs(1);
+        for _ in 0 .. retries {
```

Comment:
should this be "..=" since it's a "retries" not "tries" ?

Reply:
Maybe clearer just to rename it "tries" (or perhaps "attempts")

Reply:
yeah could be, although might be a bit to call a wait with 0 tries which won't do anything :)

Reply:
We really should use Udev::Monitor for this (i.e get notified) rather then polling in a loop with a sleep. Can you create an issue for this @cjones1024 for us to work on that in the future? 

Also, what happens if the device attaches AFTER we timedout? Do we, when we timeout, call disconnect to ensure its not there when we are done waiting?

Reply:
You've convinced me will change to a ..=

Reply:
Have created CAS-376 to use udev Monitors. Should also address appropriate failure response there as well.